common/src/test/scala/uk/ac/wellcome/sns/SNSWriterTest.scala

```diff
@@ -3,49 +3,88 @@ package uk.ac.wellcome.sns
 import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
 import org.scalatest.{FunSpec, Matchers}
 import uk.ac.wellcome.models.aws.SNSConfig
-import uk.ac.wellcome.test.fixtures.SNS
+import uk.ac.wellcome.test.fixtures._
+import uk.ac.wellcome.models.aws.S3Config
+import org.scalatest.EitherValues
 
 class SNSWriterTest
     extends FunSpec
     with ScalaFutures
     with Matchers
     with SNS
-    with IntegrationPatience {
-
-//  val topicArn = createTopicAndReturnArn("test-topic-name")
-//
+    with S3
+    with IntegrationPatience
+    with EitherValues {
 
   it(
     "should send a message with subject to the SNS client and return a publish attempt with the id of the request") {
     withLocalSnsTopic { topic =>
-      val snsConfig = SNSConfig(topic.arn)
-      val snsWriter = new SNSWriter(snsClient, snsConfig)
-      val message = "sns-writer-test-message"
-      val subject = "sns-writer-test-subject"
-      val futurePublishAttempt = snsWriter.writeMessage(
-        message = message,
-        subject = subject
-      )
-
-      whenReady(futurePublishAttempt) { publishAttempt =>
-        val messages = listMessagesReceivedFromSNS(topic)
-        messages should have size (1)
-        messages.head.message shouldBe message
-        messages.head.subject shouldBe subject
-        publishAttempt.id should be(Right(messages.head.messageId))
+      withLocalS3Bucket { bucket =>
+        val s3Config = S3Config(bucketName = bucket.name)
+        val snsConfig = SNSConfig(topic.arn)
+        val snsWriter = new SNSWriter(snsClient, snsConfig, s3Client, s3Config)
+        val message = "sns-writer-test-message"
+        val subject = "sns-writer-test-subject"
+
+        val eventualAttempt = snsWriter.writeMessage(message, subject)
+
+        whenReady(eventualAttempt) { publishAttempt =>
+          val messages = listMessagesReceivedFromSNS(topic)
+          messages should have size (1)
+          messages.head.message shouldBe bucket.name
+          messages.head.subject shouldBe subject
+          publishAttempt.right.value.id should be(messages.head.messageId)
+
+          getContentFromS3(bucket, publishAttempt.right.value.key) should be(
+            message)
+        }
+      }
+    }
+  }
+
+  it(
+    "should return a failed future if it fails to publish the message pointer") {
```

Comment:
We try and avoid `should` in test descriptions as it's redundant. The test asserts something _will_ happen and fails if it doesn't. Apologies for the pedantry.

Reply:
This set of tests looks good though.